server/services/userService.ts

```diff
@@ -1,25 +1,28 @@
 import { convertToTitleCase } from '../utils/utils'
-import type HmppsAuthClient from '../data/hmppsAuthClient'
-import type { User } from '../data/hmppsAuthClient'
-import { NomisUserRolesApi, type UserCaseloads } from '../data/nomisUserRolesApi'
+import { NomisUserRolesApi, type Caseload } from '../data/nomisUserRolesApi'
+import ManageUsersApiClient, { User } from '../data/manageUsersApiClient'
 
-export interface UserDetails extends User, UserCaseloads {
+export interface UserDetails extends User {
   displayName: string
+  caseloads: Array<Caseload>
+  activeCaseload: Caseload
 }
 
 export default class UserService {
-  constructor(private readonly hmppsAuthClient: HmppsAuthClient) {}
+  constructor(private readonly manageUsersApiClient: ManageUsersApiClient) {}
 
   async getUser(token: string): Promise<UserDetails> {
-    const nomisUserRolesApi = new NomisUserRolesApi(token)
+    return this.manageUsersApiClient.getUser(token).then(user => {
+      const nomisUserRolesApi = new NomisUserRolesApi(token)
 
-    const user = await this.hmppsAuthClient.getUser(token)
-    const userCaseloads = await nomisUserRolesApi.getUserCaseloads()
-
-    return {
-      ...user,
-      ...userCaseloads,
-      displayName: convertToTitleCase(user.name),
-    }
+      return nomisUserRolesApi.getUserCaseloads().then(uc => {
+        return {
+          ...user,
+          displayName: convertToTitleCase(user.name),
+          caseloads: uc.caseloads,
+          activeCaseload: uc.activeCaseload,
```

Comment:
these 2 lines specifically pick out 2 fields from `uc` but we could just merge in the whole thing using `...uc`

Reply:
FWIW, I like that it is explicit (in case the upstream API adds fields, which in theory is meant to be backward compatible, but in this case it may override the attributes coming from Manage Users API _**potentially**_)